components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -2299,6 +2299,20 @@ else if ("subcontext".equalsIgnoreCase(searchType)) {
         return filterMultipleVersionedAPIs(searchResults);
     }
 
+    @Override
+    public boolean isSubscribedToApp(APIIdentifier apiIdentifier, String userId, int applicationId) throws
+            APIManagementException {
+        boolean isSubscribed;
+        try {
+            isSubscribed = apiMgtDAO.isSubscribedToApp(apiIdentifier, userId, applicationId);
+        } catch (APIManagementException e) {
+            String msg = "Failed to check if user(" + userId + ") has subscribed to " + apiIdentifier;
+            log.error(msg, e);
```

Comment:
Check whether we log the same exception elsewhere.  If so, there are duplicate exceptions being logged.